### PR TITLE
Update the FFI related code intended for JEP434

### PR DESCRIPTION
**Pull request by ChengJin01**
The changes update part of our code in the FFI
framework against the latest APIs specified in
JDK20 to accommodate the new features of JEP434.

Note:
This is the part of implementation intended for
#eclipse-openj9/openj9/issues/16329

Signed-off-by: ChengJin01 <jincheng@ca.ibm.com>

### CODE REVIEW

**Comment by ChengJin01** [comment]
Note: the changes must be merged together with or after the corresponding changes in OpenJ9 at https://github.com/eclipse-openj9/openj9/pull/16835

Reviewer: @tajila, @keithc-ca
FYI: @DanHeidinga, @pshipton

**Comment by ChengJin01** [comment]
The PR was updated with the same changes against Keith's comment at https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/559.

File: src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java [hunk 1]
```diff
@@ -41,12 +41,12 @@
...
-import java.lang.foreign.GroupLayout;
-import java.lang.foreign.MemoryLayout;
-import java.lang.foreign.SegmentScope;
 import java.lang.foreign.FunctionDescriptor;
+import java.lang.foreign.GroupLayout;
 import java.lang.foreign.Linker;
+import java.lang.foreign.MemoryLayout;
 import java.lang.foreign.MemorySegment;
+import java.lang.foreign.SegmentScope;
 import java.lang.foreign.SequenceLayout;
```
Comment (keithc-ca) [inline src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java:50]: Please don't reorder these imports relative to upstream.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java (hunk 1 above)
Comment (ChengJin01) [reply src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java]: Restored as requested above.

**Comment by keithc-ca** [comment]
Jenkins compile alinux64 jdk20 depends https://github.com/eclipse-openj9/openj9/pull/16835

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**